apps/glusterfs/device_entry.go

```diff
@@ -463,3 +464,33 @@ func (d *DeviceEntry) Remove(db wdb.DB,
 func DeviceEntryUpgrade(tx *bolt.Tx) error {
 	return nil
 }
+
+func (d *DeviceEntry) DeleteBricksWithEmptyPath(tx *bolt.Tx) error {
+	godbc.Require(tx != nil)
+
+	for _, id := range d.Bricks {
+		fmt.Fprintf(os.Stdout, "before calling brick entry %v", id)
+		brick, err := NewBrickEntryFromId(tx, id)
+		if err != nil {
+			return err
+		}
+		fmt.Fprintf(os.Stdout, "before going into delete %v", id)
+		if brick.Info.Path == "" {
+			d.StorageFree(brick.TotalSize())
+			d.BrickDelete(brick.Info.Id)
+			err = d.Save(tx)
```

Comment:
should not be saving here. 
Should cache and save at the end of loop or on err

Reply:
Sounds good. Since you are planning on doing that it might be cool to also have a dry-run mode that only prints what its going to do. I don't think that would be too much extra effort.